pkg/util/push/cortexotlpconverter/metrics_to_prw.go

```diff
@@ -0,0 +1,373 @@
+package cortexotlpconverter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"sort"
+	"time"
+
+	"github.com/prometheus/otlptranslator"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/util/annotations"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/multierr"
+
+	"github.com/cortexproject/cortex/pkg/cortexpb"
+)
+
+type PromoteResourceAttributes struct {
+	promoteAll bool
+	attrs      map[string]struct{}
+}
+
+type Settings struct {
+	Namespace                         string
+	ExternalLabels                    map[string]string
```

Comment:
What's the usecase of `Namespace` and `ExternalLabels`? Do we have those configs before in Cortex? If there is no such config before, I recommend removing them if there is no such usecase